functions/Backup-DbaDatabase.ps1

```diff
@@ -453,8 +453,17 @@ function Backup-DbaDatabase {
 
             if ($CompressBackup) {
                 if ($db.EncryptionEnabled) {
-                    Write-Message -Level Warning -Message "$dbName is enabled for encryption, will not compress"
-                    $backup.CompressionOption = 2
+                    $minVerForTDECompression = [version]'13.0.4446.0' # SQL Server 2016 CU 4
+                    $flagTDESQLVersion = $minVerForTDECompression -le $Server.version
+                    $flagTestBoundMaxTransferSize = test-bound 'MaxTransferSize'
+                    $flag_correct_MaxTransferSize = $flagTestBoundMaxTransferSize -and ($MaxTransferSize -gt 65536)
+                    if ($flagTDESQLVersion -and $flagTestBoundMaxTransferSize -and $flag_correct_MaxTransferSize) {
+                        write-message -Level Verbose "$dbName is enabled for encryption but will compress"
```

Comment:
write-message -> Write-Message

Reply:
Thank you, I submitted a new commit with your suggestions.

Bye